### DISCUSSION TITLE
Raspberry Pi Bookworm using Ubuntu Lunar PPA

### DISCUSSION THREAD

**Discussion by FreaktheBear** [Stellarium/stellarium#3943] [Show and tell]
Dear Team,

I found out the following instructions to install the Ubuntu ARM Lunar version onto Raspberry Pi Bookworm (both Lunar and Bookworm share Debian 12).
I have a the latest Stellarium version working on Arm64 Ubuntu 24.04 (Raspberry Pi 4B 8GB) but have a lot of grieve accessing the ttyS0 port under Ubuntu to communicate with alien systems.
The Raspberry Pi OS is a bit more forgiven with GPIO UART access.

Instructions:
Add the following lines (see page https://answers.launchpad.net/~stellarium/+archive/ubuntu/daily "Technical details about this PPA" and choose "Lunar") either to the /etc/apt/sources.list or create a new file at e.g. /etc/apt/sources.list.d/stellarium-ppa.list:

deb https://ppa.launchpadcontent.net/stellarium/daily/ubuntu lunar main 
deb-src https://ppa.launchpadcontent.net/stellarium/daily/ubuntu lunar main 

Click the signing key under the same "Technical details about this PPA" and download the pub signing key.
Then copy the downloaded *.asc key from the Download folder to /etc/apt/trusted.gpg.d with, sudo mv e12687626702104cad5b767deb690a6429908236.asc /etc/apt/trusted.gpg.d

Then to install, perform sudo apt update, and then sudo apt install stellarium.
Run raspi-config from the terminal and activate the "X11 backend" in raspi-config, Advanced options, Wayland, X11.
(Please take note that the new Raspberry Pi Connect icon will not be visible in the Task Bar under X11 mode)
The program has to be started within the /usr/bin directory with the option "QT_QPA_PLATFORM=xcb ./stellarium --opengl-compat" else the screen remains black.
Stellarium version 2023-12-23 runs with around 8fps on a Raspberry Pi 4B 4GB (don't know if this has to do with this version).

Please let me know if this instruction might be a good idea for people who are less confident in building from source.
If so, would it be possible that the Arm64 Ubuntu Lunar version gets updated too (currently 2023-12-23) so that new functionality will work?

Kind regards

**Comment by FreaktheBear** [comment]
Updated the links to the Daily repository

**Comment by alex-w** [comment]
Important note: Stellarium for Ubuntu Lunar was builded without support GPS (without `libgps-dev`), yesterday we re-build packages for Ubuntu 22.04, 24.04 and 24.10 with GPS support.

**Comment by alex-w** [comment]
@FreaktheBear do you just need a binary package for Bookworm/arm64?

**Reply by FreaktheBear** [reply]
Hi Alex,

If you can point me to a binary package for Bookworm then that would be welcome, I will try to build it then.
The reason for my test with the Arm Lunar package was to see if I could do it with the "standard" PPA update.
Would it be possible to have Arm 23.04 (Lunar) updated too?

Kind regards

**Reply by alex-w** [reply]
> Would it be possible to have Arm 23.04 (Lunar) updated too?

No, we can't update PPA for 23.04 due limitations of launchpad.net

**Reply by FreaktheBear** [reply]
Okay, no worries.

**Reply by FreaktheBear** [reply]
Hi Alex,
I changed the ppa to point to "jammy":

deb https://ppa.launchpadcontent.net/stellarium/daily/ubuntu jammy main
deb-src https://ppa.launchpadcontent.net/stellarium/daily/ubuntu jammy main

This also works with the instructions I mentioned earlier for the Rpi4B running with bookworm.
GPS is working and refresh is around 10fps.
I am happy.

Kind regards
